### PR TITLE
WT-5581 Fix memory leaks

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: test/csuite/wt4105_large_doc_small_upd/main.c [hunk 1]
```diff
@@ -133,7 +133,7 @@ main(int argc, char *argv[])
             modify_entry.offset = offset;
             modify_entry.size = modify_entry.data.size;
             /* FIXME-PM-1521: extend timeout to pass the test */
-            (void)alarm(5);
+            (void)alarm(7);
```
Comment (quchenhao) [inline test/csuite/wt4105_large_doc_small_upd/main.c]: Not sure why overhead increases and I have to extend the timeout further to pass this test.

**Comment by quchenhao** [comment]
evergreen retry

**Comment by quchenhao** [comment]
@kommiharibabu Can you have a look at this?

File: src/btree/bt_cursor.c [hunk 2]
```diff
@@ -379,6 +381,8 @@ __wt_cursor_valid(WT_CURSOR_BTREE *cbt, WT_UPDATE **updp, bool *valid)
             }
             if (updp != NULL)
                 *updp = upd;
+            else if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DISK))
```
Comment (kommiharibabu) [inline src/btree/bt_cursor.c]: Doesn't it need the same fix for `BTREE_COL_VAR` page type also?

Comment (quchenhao) [reply src/btree/bt_cursor.c]: Oh, correct.

Comment (quchenhao) [reply src/btree/bt_cursor.c]: Fixed

**Pull request merged**